File: 136-single-number/single-number.java
class Solution {
    public static int singleNumber(int[] nums) {
        int result = 0;

        // XOR all elements
        for (int num : nums)
        {
            result = result ^ num;
        }
        return result;
    }

    public static void main(String[] args) {

        int[] nums1 = {2, 2, 1};
        int[] nums2 = {4, 1, 2, 1, 2};
        int[] nums3 = {1};

        System.out.println("Output: " + singleNumber(nums1));
        System.out.println("Output: " + singleNumber(nums2));
        System.out.println("Output: " + singleNumber(nums3));
    }
}

File: 136-single-number/single-number.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        # XOR all elements
        for num in nums:
            result = result ^ num
        return result

# Create object
obj = Solution()

# Example Inputs
nums1 = [2, 2, 1]
nums2 = [4, 1, 2, 1, 2]
nums3 = [1]

# Print Outputs
print(obj.singleNumber(nums1))  # 1
print(obj.singleNumber(nums2))  # 4
print(obj.singleNumber(nums3))  # 1